src/plugins/undoRedo/undoRedo.js

```diff
@@ -428,39 +427,48 @@ UndoRedo.RemoveColumnAction.prototype.undo = function(instance, undoneCallback)
   const ascendingIndexes = this.indexes.slice(0).sort();
   const sortByIndexes = (elem, j, arr) => arr[this.indexes.indexOf(ascendingIndexes[j])];
 
+  const removedDataLength = this.data.length;
   const sortedData = [];
-  rangeEach(this.data.length - 1, (i) => {
-    sortedData[i] = arrayMap(this.data[i], sortByIndexes);
-  });
 
-  let sortedHeaders = [];
-  sortedHeaders = arrayMap(this.headers, sortByIndexes);
+  for (let rowIndex = 0; rowIndex < removedDataLength; rowIndex++) {
+    sortedData.push(arrayMap(this.data[rowIndex], sortByIndexes));
+  }
 
+  const sortedHeaders = arrayMap(this.headers, sortByIndexes);
+  const isFormulaPluginEnabled = instance.getPlugin('formulas') ? instance.getPlugin('formulas').enabled : false;
   const changes = [];
 
   instance.alter('insert_col', this.indexes[0], this.indexes.length, 'UndoRedo.undo');
 
-  rangeEach(this.data.length - 1, (i) => {
-    const row = instance.getSourceDataAtRow(i);
+  arrayEach(instance.getSourceDataArray(), (rowData, rowIndex) => {
```

Comment:
What about native `Array.each`?

Reply:
`arrayEach` is more stable :) There is no policy about the usage of the native vs helpers. I'll leave as it is.